### PR TITLE
✨ Add support for Python 3.10

### PR DESCRIPTION
**Pull request by tiangolo**
✨ Add support for Python 3.10

This will have to wait until Uvicorn requires httptools `0.3.0` with support for Python 3.10 https://github.com/MagicStack/httptools/releases/tag/v0.3.0

### CODE REVIEW

**Comment by hingston** [comment]
#111 will allow this to build. Missing gcc for python-slim image

**Comment by skraus-dev** [comment]
Is it an option to pin `httptools` version requirement ourselves to `httptools>=0.3.0` ?

Uvicorn defines it in such boundaries where this should possible.

https://github.com/encode/uvicorn/blob/efdbe7a4a6c94f34c35d751fb63391136b4e90e4/setup.py#L54-L57

```
extra_requirements = [
    "websockets>=10.0",
    "httptools>=0.2.0,<0.4.0",
    "uvloop>=0.14.0,!=0.15.0,!=0.15.1; " + env_marker_cpython,
```

**Comment by thiagosalvatore** [comment]
Any updates? I'm planning to update my app to use python 3.10 and this release would help a lot

**Comment by SionAbes** [comment]
This looks like it can be re-run, though an update to 0.17.6 as [mentioned by dependabot](https://github.com/tiangolo/uvicorn-gunicorn-docker/pull/127) would be needed! Looking forward to the performance increases on MacOS! ✨⚡️

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -20,7 +22,9 @@ jobs:
             python_version: "3.7"
           - name: python3.6
             python_version: "3.6"
-          - name: python3.9-slim
+          - name: python3.10-slim
+            python_version: "3.9"
+          - name: python3.10-slim
             python_version: "3.9"
```
Comment (cdgriffith) [inline .github/workflows/deploy.yml:28]:
```suggestion
          - name: python3.10-slim
            python_version: "3.10"
          - name: python3.9-slim
            python_version: "3.9"
```

Maybe I'm missing my morning caffeine, but I think this is what you meant?

**Comment by sylvainmouquet** [comment]
the project is still maintained ?

**Comment by AlTosterino** [comment]
Just bumping 🙌

**Comment by tienhm0202** [comment]
Bump as well

**Comment by Charlie-iProov** [comment]
bump!

**Comment by bazylhorsey** [comment]
bump!

**Comment by karamfil** [comment]
Hey @tiangolo, is there a timeline for this change?

**Comment by agates4** [comment]
Looking forward to this change

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (tiangolo) [reply .github/workflows/deploy.yml]: That's right, thank you!

**Comment by tiangolo** [comment]
Done! :rocket:  :tada: 

Thanks for the input everyone! :cake:

**Pull request merged**